corehq/apps/es/users.py

```diff
@@ -217,14 +229,29 @@ def is_practice_user(practice_mode=True):
 
 
 def role_id(role_id):
-    return filters.OR(
-        filters.term("domain_membership.role_id", role_id),     # mobile users
-        filters.term("domain_memberships.role_id", role_id)     # web users
+    return queries.nested(
+        'user_domain_memberships',
+        filters.term('user_domain_memberships.role_id', role_id),
     )
 
 
-def is_active(active=True):
-    return filters.term("is_active", active)
+def is_active(active=True, domain=None):
```

Comment:
Is there a need to use the `is_active` filter without passing in a domain?  I worry that making optional makes it too easy to forget to support web users